pkg/kncloudevents/message_sender_test.go

```diff
@@ -99,16 +224,20 @@ func TestRetryConfigFromDeliverySpec(t *testing.T) {
 			}
 
 			// Create the RetryConfig from the deliverySpec
-			retryConfig, err := RetryConfigFromDeliverySpec(deliverySpec)
+			retryConfig, err := RetryConfigFromDeliverySpecWith429(deliverySpec, tc.respectRetryAfter)
 			assert.Equal(t, tc.wantErr, err != nil)
 
 			// If successful then validate the retryConfig (Max & Backoff calculations).
 			if err == nil {
 				assert.Equal(t, retry, retryConfig.RetryMax)
 				for i := 1; i < retry; i++ {
 					expectedBackoffDuration := tc.expectedBackoffDurations[i-1]
-					actualBackoffDuration := retryConfig.Backoff(i, nil)
-					assert.Equal(t, expectedBackoffDuration, actualBackoffDuration)
+					actualBackoffDuration := retryConfig.Backoff(i, tc.response)
+					if tc.roundExpectedDuration > 0 {
+						assert.Equal(t, expectedBackoffDuration, actualBackoffDuration.Round(tc.roundExpectedDuration)) // Round Time String Cases
+					} else {
+						assert.Equal(t, expectedBackoffDuration, actualBackoffDuration)
+					}
```

Comment:
This may be trivial but Duration.Round(0) always returns the original duration ("If m <= 0, Round returns d unchanged."), so this extra logic is unnecessary
```suggestion
						assert.Equal(t, expectedBackoffDuration, actualBackoffDuration.Round(tc.roundExpectedDuration)) // Round Time String Cases
```

Reply:
👍